### DISCUSSION TITLE
Retrieve rules from database

### DISCUSSION THREAD

**Discussion by tlaraMQ** [hyperjumptech/grule-rule-engine#469] [Q&A]
Hello everyone,

I'd like to open this topic again. Grule's FAQs state that they are not implementing any adapter to a database, however, they state it is a good idea and that it is possible to do this with the available tools.  I'd like to ask the community if someone has implemented saving Grule rules into some kind of database and then retrieving them from it into Grule's execution. By now I only know that it is possible to load rules from bytes and .grl files (the last one I've used the most).

I'd appreciate any insights regarding this matter.
Best regards,

**Comment by davidsonff** [comment]
We're doing it from Github repos.

**Reply by cloudvant** [reply]
```
func (rb *RulesEngine) CreateRulesKB() *ast.KnowledgeBase {
    rules, _ := DB.Rules.Get()
    lib := ast.NewKnowledgeLibrary()
    builder := builder.NewRuleBuilder(lib)
    for _, r := range *rules {
       err := builder.BuildRuleFromResource(rb.Name, rb.Version, pkg.NewBytesResource([]byte(r.Text)))
       if err != nil {
          fmt.Println( fmt.Errorf("builder.BuildRuleFromResource: %v", err))
       }
    }
    kb, _ := lib.NewKnowledgeBaseInstance(rb.Name, rb.Version)
    return kb
}
```
